client/src/util/stateManager/world.js

```diff
@@ -38,80 +37,55 @@ Notable keys in the world object:
   A dataframe containing the X/Y layout for all obs.  Columns are named
   'X' and 'Y', and rows are indexed in the same way as obsAnnotation.
 
-* summary: summary of each obsAnnotation column (eg, numeric extent for
-  continuous data, category counts for categorical metadata)
-
-* varDataCache: expression columns, in a kvCache.   TODO: maybe move to a
-  Dataframe in the future.
+* varData: a cache of expression columns, stored in a Dataframe.  Cache
+  managed by controls reducer.
 
 */
 
-/* varDataCache config - see kvCache for semantics */
-const VarDataCacheLowWatermark = 32; // cache element count
-const VarDataCacheTTLMs = 1000; // min cache time in MS
-
 function templateWorld() {
   return {
     /* schema/version related */
-    api: null,
     schema: null,
     nObs: 0,
     nVar: 0,
 
     /* annotations */
-    obsAnnotations: null,
-    varAnnotations: null,
+    obsAnnotations: Dataframe.Dataframe.empty(),
+    varAnnotations: Dataframe.Dataframe.empty(),
 
     /* layout of graph. Dataframe. */
-    obsLayout: null,
-
-    /* derived data summaries XXX: consider exploding in place */
-    summary: null,
+    obsLayout: Dataframe.Dataframe.empty(),
 
-    varDataCache: kvCache.create(
-      VarDataCacheLowWatermark,
-      VarDataCacheTTLMs
-    ) /* cache of var data (expression) */
+    /*
+    Var data columns - subset of all data (may be empty)
+    */
+    varData: Dataframe.Dataframe.empty(null, new Dataframe.KeyIndex())
   };
 }
 
 export function createWorldFromEntireUniverse(universe) {
-  if (!universe.finalized) {
-    throw new Error("World can't be created from an partial Universe");
-  }
-
   const world = templateWorld();
 
   /*
   public interface follows
   */
 
   /* Schema related */
-  world.api = universe.api;
   world.schema = universe.schema;
   world.nObs = universe.nObs;
   world.nVar = universe.nVar;
 
-  /* annotations */
+  /* annotation dataframes */
   world.obsAnnotations = universe.obsAnnotations;
   world.varAnnotations = universe.varAnnotations;
 
-  /* layout and display characteristics */
+  /* layout and display characteristics dataframe */
   world.obsLayout = universe.obsLayout;
 
-  /* derived data & summaries */
-  world.summary = summarizeAnnotations(
-    world.schema,
-    world.obsAnnotations,
-    world.varAnnotations
-  );
-
-  /* build the varDataCache */
-  world.varDataCache = kvCache.map(
-    universe.varDataCache,
-    val => subsetVarData(world, universe, val),
-    { lowWatermark: VarDataCacheLowWatermark, minTTL: VarDataCacheTTLMs }
-  );
+  /*
+  Var data columns - subset of all
+  */
+  world.varData = universe.varData.clone();
```

Comment:
Subset === clone?

Reply:
nope.

clone === shallow copy of entire dataframe.
cutByList, icutByList, icutByMask === subset/slice

Reply:
@colinmegill - per offline conversation, the "cut" family of API has been renamed to "subset".  Please give it a look.